src/algebra/monoid_algebra/grading.lean

```diff
@@ -182,31 +182,39 @@ graded_algebra.of_alg_hom _
   end)
   (λ i x, by convert (decompose_aux_coe f x : _))
 
+-- Lean can't find this later without us repeating it
+instance grade_by.decomposition : direct_sum.decomposition (grade_by R f) :=
```

Comment:
idem: Is this a known issue (the same one as the missing finsupp instances)? Or could you add a bit more explanation why it breaks?